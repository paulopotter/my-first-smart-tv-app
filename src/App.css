body {
  background-color: #0f0f0f;
  color: #f1f1f1;
}

.poster-wrapper {
  display: inline-block;
  height: 390px;
  text-align: center;
  vertical-align: bottom;
  width: 310px;
}

.poster-wrapper figure {
  position: relative;
}

.poster-wrapper img {
  height: 321px;
  width: 225px;
}

.poster-wrapper--active img{
  height: 381px;
  width: 285px;
}

.poster-wrapper span {
  display: none;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: #000;
  min-height: 15%;
  padding: 15px 10px;
  font-size: 20px;
}

.poster-wrapper--active span {
  display: inline;
}
