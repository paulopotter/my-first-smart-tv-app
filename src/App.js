/* eslint-disable camelcase */
import React from 'react'
import jikanjs from 'jikanjs'

import './App.css'

class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      animes: [],
      activeItem: 0,
    }
    this.onKeyDown = this.onKeyDown.bind(this)
  }

  componentDidMount() {
    const animes = []
    jikanjs
      .loadSeasonLater()
      .then(response => {
        response.anime.forEach(element => {
          animes.push(element)
        })
        this.setState({ animes })
      })
      .catch(err => {
        window.console.error(err) // in case a error happens
      })

    this.enableKeyEvent()
  }

  componentWillUnmount() {
    this.disableKeyEvent()
  }

  enableKeyEvent() {
    document.addEventListener('keydown', this.onKeyDown, true)
  }

  disableKeyEvent() {
    document.removeEventListener('keydown', this.onKeyDown, true)
  }

  onKeyDown(e) {
    const { activeItem, animes } = this.state
    console.log('->', animes.length)
    let newActiveItem = activeItem
    switch (e.keyCode) {
      case 37: // left
        if (activeItem !== 0) {
          newActiveItem -= 1
        }
        break

      case 39: // right
        if (activeItem < animes.length) {
          newActiveItem += 1
        }
        break

      case 40: // down
        if (activeItem < animes.length) {
          if (activeItem + 4 > animes.length) {
            newActiveItem = animes.length
          } else {
            newActiveItem += 4
          }
        }
        break
      case 38: // up
        if (activeItem < animes.length) {
          if (activeItem - 4 < 0) {
            newActiveItem = 0
          } else {
            newActiveItem -= 4
          }
        }
        break
      default:
        break
    }
    this.setState({
      activeItem: newActiveItem
    })
  }

  tmpl(anime, i) {
    const { image_url, title, mal_id } = anime
    const { activeItem } = this.state

    return (
      <div
        className={`poster-wrapper ${
          activeItem === i ? 'poster-wrapper--active' : null
        }`}
        id={mal_id}
        key={mal_id}
      >
        <figure>
          <img src={image_url} alt={title} />
          <p>{title}</p>
        </figure>
      </div>
    )
  }

  render() {
    const { animes } = this.state
    console.log( this.state.activeItem )
    return animes.map((anime, i) =>
      anime.type === 'TV' ? this.tmpl(anime, i) : null
    )
  }
}

export default App
