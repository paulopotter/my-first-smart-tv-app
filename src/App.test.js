import React from 'react';
import { shallow, mount, render } from 'enzyme';
import App from './App';

describe('App', () => {
  it('Should match snapshot', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot()
  });

  it('renders without crashing', () => {
    const component = mount(<App />);
    expect(component).toHaveLength(1);
  });
})

