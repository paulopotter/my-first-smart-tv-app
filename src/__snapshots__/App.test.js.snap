// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `ShallowWrapper {}`;
